packages/nvidia_nat_memmachine/src/nat/plugins/memmachine/memory.py

```
@@ -0,0 +1,114 @@
+    """
+    base_url: str  # Base URL of the MemMachine server (e.g., "http://localhost:8080")
+    org_id: str | None = None  # Optional default organization ID
+    project_id: str | None = None  # Optional default project ID
+    timeout: int = 30  # Request timeout in seconds
+    max_retries: int = 3  # Maximum number of retries for failed requests
+
+
+@register_memory(config_type=MemMachineMemoryClientConfig)
+async def memmachine_memory_client(
+    config: MemMachineMemoryClientConfig,
+    builder: Builder,  # Required by @register_memory contract, unused here
```

Ruff (0.14.13)

Unused function argument: `builder`

(ARG001)